File: Basics-and-Pillars/TypeScript/src/Employee.ts
import { Person } from './Person';

export class Employee extends Person {
    private salary: number;

    constructor(name: string, age: number, salary: number) {
        super(name, age);
        this.salary = salary;
    }

    public work(): void {
        console.log(`${this.getName()} is working...`);
    }

    public override introduce(): void {
        console.log(`Employee: ${this.getName()}`);
    }
}

File: Basics-and-Pillars/Java/src/Employee.java
public class Employee extends Person {

    private double salary;

    public Employee(String name, int age, double salary) {
        super(name, age);
        this.salary = salary;
    }

    public void work() {
        System.out.println(getName() + " is working...");
    }

    @Override
    public void introduce() {
        System.out.println("Employee: " + getName());
    }
}
